thewalrus/_hafnian.py

```diff
@@ -178,6 +180,45 @@ def hafnian(
     )
 
 
+def hafnian_sparse(A, D = None):
+    r"""Returns the hafnian of a sparse symmetric matrix.
+    This pure python implementation is very slow on full matrices, but faster the sparser a matrix is.
+    As a rule of thumb, the crossover in runtime with respect to hafnian() happens around 50% sparsity.
```

Comment:
Should add a ref to the `hafnian` func
```suggestion
    As a rule of thumb, the crossover in runtime with respect to :func:`~.hafnian` happens around 50% sparsity.
```